src/CloudNative.CloudEvents.NewtonsoftJson/JsonEventFormatter.cs

```diff
@@ -141,16 +145,15 @@ private CloudEvent DecodeJObject(JObject jObject, IEnumerable<CloudEventAttribut
             {
                 throw new ArgumentException($"Structured mode content does not represent a CloudEvent");
             }
-            var specVersion = CloudEventsSpecVersion.FromVersionId((string) specVersionToken);
-            if (specVersion is null)
-            {
-                throw new ArgumentException($"Unsupported CloudEvents spec version '{(string)specVersionToken}'");
-            }
+            var specVersion = CloudEventsSpecVersion.FromVersionId((string) specVersionToken)
+                ?? throw new ArgumentException($"Unsupported CloudEvents spec version '{(string)specVersionToken}'");
 
             var cloudEvent = new CloudEvent(specVersion, extensionAttributes);
             PopulateAttributesFromStructuredEvent(cloudEvent, jObject);
             PopulateDataFromStructuredEvent(cloudEvent, jObject);
-            return cloudEvent.Validate();
+            // "data" is always the parameter from the public method. It's annoying not to be able to use
+            // nameof here, but this will give the appropriate result.
```

Comment:
Sorry, but why can't nameof be used? The parameter in this method is `data`.

Reply:
Unfortunately not - GitHub was hiding the fact that this code is in the `DecodeJObject` method :(